src/components/alert/index.js

```diff
@@ -0,0 +1,43 @@
+import { Alert } from "bootstrap";
+import BaseComponent from "../base-component";
+
+export default class CustomAlert extends BaseComponent {
+  timerId;
+  constructor(type = "success", message = "") {
+    super();
+    this.type = type;
+    this.message = message;
+
+    this.init();
+
+    // NOTE: register alert for closing
+    new Alert(this.element);
+
+    this.timerId = setTimeout(() => {
+      this.close();
+    }, 2000);
+  }
+
+  close() {
+    clearTimeout(this.timerId);
+
+    const alert = Alert.getInstance(this.element);
+
+    if (alert) {
+      alert.close();
+    }
+  }
+
+  get template() {
+    return `
+      <div class="alert alert-${this.type} alert-dismissible fade show fixed-top w-100" role="alert" data-cy="alert">
```

Comment:
**Security Issue**: The template string in 'get template()' method is directly embedding user-controlled input ('this.type' and 'this.message'), which can lead to Cross-Site Scripting (XSS) if the input is not properly sanitized. <br> **Fix**: Ensure that 'this.type' and 'this.message' are properly sanitized or escaped to prevent XSS. <br> **Code Suggestion**: 
 ```
 - <div class="alert alert-${this.type} alert-dismissible fade show fixed-top w-100" role="alert" data-cy="alert">
 + <div class="alert alert-${encodeHTML(this.type)} alert-dismissible fade show fixed-top w-100" role="alert" data-cy="alert">
 - ${this.message}
 + ${encodeHTML(this.message)}
 ```